pkg/plugins/golang/v3/scaffolds/internal/templates/readme.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+import "sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+
+var _ machinery.Template = &Readme{}
+
+// Readme scaffolds a README.md file
+type Readme struct {
+	machinery.TemplateMixin
+	machinery.BoilerplateMixin
+	machinery.ProjectNameMixin
+}
+
+// SetTemplateDefaults implements file.Template
+func (f *Readme) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = "README.md"
+	}
+
+	f.TemplateBody = readmeFileTemplate
+
+	return nil
+}
+
+//nolint:lll
+const readmeFileTemplate = `# {{ .ProjectName }}
+// TODO(user): Add simple overview of use/purpose
+
+## Description
+// TODO(user): An in-depth paragraph about your project and overview of use
+
+## Getting Started
+You’ll need a Kubernetes cluster to run against. You can use [KIND](https://sigs.k8s.io/kind) to get a local cluster for testing, or run against a remote cluster.
+**Note:** Your controller will automatically use the current context in your kubeconfig file (i.e. whatever cluster ` + "`kubectl cluster-info`" + ` shows).
+
+### Running on the cluster
+1. Install Instances of Custom Resources:
+` +
+	"\t```sh\n" +
```

Comment:
why we are adding the \t? and the n"
Why does the shell needs to be inside of `""`?
Why can we not have a const as in the other places without these characters?

Reply:
> why we are adding the \t? and the n"  

I added the \t to have the code fences tabbed in underneath the list items, however now that I think about it that doesn't seem like that big of a deal so I removed them.

> Why does the shell needs to be inside of `""`?

The shell needed to be inside of `""` so that we could use the backtick to create the code fences. There doesn't seem to be a way to escape the " ` " character when using a string literal.

> Why can we not have a const as in the other places without these characters?

I thought that this may make it a bit more readable but I will go ahead and make a change that abstracts them from being put in the way they are above.

Reply:
Changes made in commit https://github.com/kubernetes-sigs/kubebuilder/pull/2610/commits/a0faa6f08895abaf1bad432af586e49ef8ce107e

Let me know what you think

Reply:
It shows sorted out so I am closing this one.